### PR TITLE
Add Configuration#use_exception_level_filters option

### LINKED ISSUE #587: allow configuring exception level filters:true/false as a global setting for manual reporting
Hi there, 
Currently if we manually report an issue through Rollbar.error(e), if we want Rollbar to take into consideration our exception level filters, we need to add a silly flag (:use_exception_level_filters => true).

Either cancel this flag, or allow for us to set it globally as a rollbar configuration setting. 

This issue just creates dirty code on our end..

### PR DESCRIPTION
**Pull request by jondeandres**
Let the users to set a global `use_exception_level_filters` option

Usage:

```
Rollbar.configure do |config|
  config.use_exception_level_filters = true
end
```

By default the setting is `false` so we keep the current interface

Fix #587

### CODE REVIEW

**Comment by rokob** [comment]
This seems fine to me. I guess the only question is whether you want this global configuration setting to be overridable on a per-call basis, i.e. if `use_exception_level_filters` is true in the configuration, based on this code, you cannot add `:use_exception_level_filters => false` on a particular error call and expect that to work. I think that is okay and conforms to what I would expect by setting the global flag. So I'd say go  ahead and merge this, once you fix the tests.

**Comment by brianr** [comment]
Hm I think I would expect that:

- the new flag @jondeandres added is the default
- the default can be overridden by passing a param in the call

What if we do something like:

- rename the new config param to `use_exception_level_filters_default`. Its default will be `false` to maintain the current behavior.
- if you don't pass the `:use_exception_level_filters` param in a call, then the value of the config param is used. (maybe do this by having the param default to `nil`)
- if you do pass the `:use_exception_level_filters` param, then the value you provide is used

**Comment by jondeandres** [comment]
@brianr @rokob I've pushed force here with Brian's suggestions.

File: spec/rollbar_spec.rb [hunk 1]
```diff
@@ -536,6 +536,82 @@
...
+        it 'sends the correct filtered level' do
+          Rollbar.configure do |config|
+            config.exception_level_filters = { 'NameError' => 'warning' }
+          end
+
+          Rollbar.error(exception)
+
+          expect(Rollbar.last_report[:level]).to be_eql('warning')
+        end
+
```
Comment (rokob) [inline spec/rollbar_spec.rb]: This is missing a test when `config.use_exception_level_filters_default = true` has been set, and when `:use_exception_level_filters => false` is passed in with the `Rollbar.error` call.

**Comment by rokob** [comment]
I squashed those commits.

File: spec/rollbar_spec.rb (hunk 1 above)
Comment (rokob) [inline spec/rollbar_spec.rb]: I added one

**Pull request merged**